rfc/system/RFC-4570-flyte-auth-in-task.md

```diff
@@ -0,0 +1,145 @@
+# [RFC Template] Flyte Authentication in Tasks
+
+**Authors:**
+
+- @zeryx
+
+## 1 Executive Summary
+
+This RFC proposes a method to flow Flyte authentication information into task containers. 
+It aims to simplify access to execution-specific information and enable secure, streamlined usage of Flyte features within tasks. 
+The proposal introduces a Flyte Auth Agent system to bind the authentication context of the task/workflow executor to all executed tasks.
+
+## 2 Motivation
+
+- In many workflows, accessing execution-specific information (e.g., used images, versions) is crucial. 
+- Tasks often need to interact with Flyte for remote registration or to use Flyte features (e.g., pyflyte register, build, project creation).
```

Comment:
I would slightly challenge this that "often" might be an overstatement here. However, I do see a fair usecase with `@eager` mode